src/state.ts

```diff
@@ -190,12 +196,18 @@ export class EditorState {
     return instance
   }
 
-  /// Create a new state based on this one, but with an adjusted set
-  /// of active plugins. State fields that exist in both sets of
-  /// plugins are kept unchanged. Those that no longer exist are
-  /// dropped, and those that are new are initialized using their
-  /// [`init`](#state.StateField.init) method, passing in the new
-  /// configuration object..
+  /**
+   * Creates a new state based on the current state with an adjusted set of active plugins.
+   * 
+   * State fields that exist in both sets of plugins are kept unchanged. 
+   * Those that no longer exist are dropped, and those that are new are
+   * initialized using their [`init`](#state.StateField.init) method,
+   * passing in the new configuration object.
+   * 
+   * @param {Object} config - Configuration object for the new state.
```

Comment:
This type of JSDoc could be updated to have a link to the Typescript Type and also be extended to show properties within the object -- though a bit redundant with TS.